packages/gatsby/src/schema/schema.js

```diff
@@ -285,10 +272,14 @@ function createChildrenField(typeName) {
   return {
     [_.camelCase(`children ${typeName}`)]: {
       type: () => [typeName],
-      resolve(source, args, context) {
-        return source.children
-          .map(context.nodeModel.getNode)
-          .filter(nodeIsOfType(typeName))
+      async resolve(source, args, context) {
+        const { path } = context
+        const result = await Promise.all(
+          source.children.map(id =>
+            context.nodeModel.getNodeByType({ id, type: typeName }, { path })
+          )
+        )
+        return result.filter(Boolean)
```

Comment:
Why can it have non-true values? getNodeByType with typename should filter the stuff.